{"name": "test_login_invalid_credentials", "status": "broken", "statusDetails": {"message": "test_loginn.InvalidCredentialsError: Invalid username or password by manual check.", "trace": "def test_login_invalid_credentials():\n        # try:\n>       login(\"wronguser@example.com\", \"wrong_password\",\n                  \"https://velvet.webredirect.himshang.com.np/#/pages/dashboard\")\n\ntest_loginn.py:71: \n_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _\n\nusername = 'wronguser@example.com', password = 'wrong_password'\nurl = 'https://velvet.webredirect.himshang.com.np/#/pages/dashboard'\n\n    @allure.step(\"Perform login with username: {username} and password: {password}\")\n    def login(username, password, url):\n        driver.get(url)\n    \n        try:\n            with allure.step(\"Filling login form\"):\n                WebDriverWait(driver, 10).until(\n                    EC.presence_of_element_located((By.CSS_SELECTOR, 'input[formcontrolname=\"username\"]'))\n                ).send_keys(username)\n    \n                driver.find_element(By.CSS_SELECTOR, 'input[formcontrolname=\"password\"]').send_keys(password)\n    \n                sign_in_btn = driver.find_element(By.XPATH, \"//button[contains(text(), 'Sign In')]\")\n                sign_in_btn.click()\n    \n            time.sleep(3)  # Wait for page to potentially show errors\n    \n            # Manual logic check\n            if (password != \"Tebahal1!\" or username != \"gedehim917@decodewp.com\"):\n                allure.attach(driver.get_screenshot_as_png(), name=\"invalid_credentials\",\n                              attachment_type=allure.attachment_type.PNG)\n>               raise InvalidCredentialsError(\"Invalid username or password by manual check.\")\nE               test_loginn.InvalidCredentialsError: Invalid username or password by manual check.\n\ntest_loginn.py:40: InvalidCredentialsError"}, "steps": [{"name": "Perform login with username: 'wronguser@example.com' and password: 'wrong_password'", "status": "broken", "statusDetails": {"message": "test_loginn.InvalidCredentialsError: Invalid username or password by manual check.\n", "trace": "  File \"C:\\Users\\karun\\AppData\\Roaming\\Python\\Python313\\site-packages\\allure_commons\\_allure.py\", line 202, in impl\n    return func(*a, **kw)\n  File \"C:\\Users\\karun\\PycharmProjects\\PythonProject\\extra testing\\Custom Exception\\ tests\\test_loginn.py\", line 40, in login\n    raise InvalidCredentialsError(\"Invalid username or password by manual check.\")\n"}, "steps": [{"name": "Filling login form", "status": "passed", "start": 1751629417687, "stop": 1751629443693}], "attachments": [{"name": "invalid_credentials", "source": "305746a0-38cc-4304-9896-37e2bf195335-attachment.png", "type": "image/png"}], "parameters": [{"name": "username", "value": "'wronguser@example.com'"}, {"name": "password", "value": "'wrong_password'"}, {"name": "url", "value": "'https://velvet.webredirect.himshang.com.np/#/pages/dashboard'"}], "start": 1751629417263, "stop": 1751629446889}], "start": 1751629417262, "stop": 1751629446893, "uuid": "265ab466-2f78-43ec-9075-9cda08cd0d1b", "historyId": "1a5a83e12672511b17434ca0c1e05048", "testCaseId": "1a5a83e12672511b17434ca0c1e05048", "fullName": "test_loginn#test_login_invalid_credentials", "labels": [{"name": "suite", "value": "test_loginn"}, {"name": "host", "value": "DESKTOP-AOJNATO"}, {"name": "thread", "value": "13412-MainThread"}, {"name": "framework", "value": "pytest"}, {"name": "language", "value": "cpython3"}, {"name": "package", "value": "test_loginn"}]}